courses-rest/src/test/java/no/sikt/nva/fs/CoursesByInstitutionOfLoggedInUserHandlerTest.java

```diff
@@ -1,65 +1,284 @@
+    public static final Clock BEFORE_SUMMER = Clock.fixed(Instant.parse("2022-03-10T00:00:00.00Z"),
+                                                          TimeProvider.ZONE_ID);
+    private static final String EXAMPLE_COM_URI = "http://example.com";
+    private static final String NON_SUPPORTED_INSTITUTION_PATH = "100.0.0.0.0";
+    private static final String SUPPORTED_INSTITUTION_PATH = "215.0.0.0.0";
+    private static final String TAUGHT_COURSES_URL_PATH = "/undervisning";
+    private static final String DB_ID_QUERY_PARAM_NAME = "dbId";
+    private static final String INSTITUTION_QUERY_PARAM_NAME = "emne.institusjon";
+    private static final String LIMIT_QUERY_PARAM_NAME = "limit";
+    private static final String YEAR_QUERY_PARAM_NAME = "semester.ar";
+    private static final String SUCCESS_AUTHORIZATION_HEADER_VALUE = "Basic ZHVtbXlVc2VybmFtZTpkdW1teVBhc3N3b3Jk";
+    private static final String APPLICATION_JSON_CONTENT_TYPE_VALUE = "application/json";
+    private static final Course COURSE_B_SPRING_2022 = new Course("B", "V\u00C5R", 2022); // VÅR
+    private static final Course COURSE_OE_SPRING_2022 = new Course("\u00D8", "V\u00C5R", 2022); // Ø, VÅR
+    private static final Course COURSE_AA_SPRING_2022 = new Course("\u00C5", "V\u00C5R", 2022); // Å, VÅR
+    private static final Course COURSE_A_AUTUMN_2022 = new Course("A", "H\u00D8ST", 2022); // HØST
+    private static final Course COURSE_AE_AUTUMN_2022 = new Course("\u00C6", "H\u00D8ST", 2022); // Æ, HØST
+    private static final Course COURSE_A_SPRING_2023 = new Course("A", "V\u00C5R", 2023); // VÅR
+    private static final Course COURSE_OE_SPRING_2023 = new Course("\u00D8", "V\u00C5R", 2023); // Ø, VÅR
+    private static final Course COURSE_AA_SPRING_2023 = new Course("\u00C5", "V\u00C5R", 2023); // Å, VÅR
+    final Environment environment = mock(Environment.class);
     private Context context;
     private CoursesByInstitutionOfLoggedInUserHandler handler;
     private ByteArrayOutputStream output;
 
     @BeforeEach
-    public void init() {
-        var environment = mock(Environment.class);
+    public void init(final WireMockRuntimeInfo wmRuntimeInfo) {
         when(environment.readEnv(ApiGatewayHandler.ALLOWED_ORIGIN_ENV)).thenReturn("*");
 
-        context = mock(Context.class);
+        final String fsBaseUri = UriWrapper.fromUri(wmRuntimeInfo.getHttpBaseUrl()).toString();
+        final String fsConfigString = IoUtils.stringFromResources(Path.of("fsConfig.json"))
+                                          .replace("@@BASE_URI@@", fsBaseUri);
+        when(environment.readEnvOpt(CoursesByInstitutionOfLoggedInUserHandler.FS_CONFIG_ENV_NAME))
+            .thenReturn(Optional.of(fsConfigString));
 
+        context = mock(Context.class);
         output = new ByteArrayOutputStream();
+    }
+
+    @Test
+    void shouldReturnEmptyListOfCoursesAndLogProblemIfFsIsUnavailable() throws IOException {
```

Comment:
Why are we not returning BadGateway when FS is not available?

Reply:
Fixed